### ISSUE TITLE
sklearn handles text labels differently than ml_dask on OneHotEncoding

### ISSUE THREAD

**Issue by brian-methodical** [dask/dask-ml#964]
**Describe the issue**:
[OneHotEncoding](https://github.com/dask/dask-ml/blob/ba2024d68557fade43f4dc2fa369445d01629923/dask_ml/preprocessing/_encoders.py#L16) defaults to sets `catagories_` to a numpy dtype='<U1' when explicitly given an array of text [["a", "b", "c"]] sklearn sets `catagories_` dtype `object`.
Which is the correct behavior:

1. either cast `catagories_` to dtype=object from OneHotEncoding on dask-ml when explicitly setting arrays (auto[1] is off)
2. or fix the test so it does not care about `catagories_` dtype mismatch OR explicitly set dtype as argument in tests

[1] please note this issue is only with categories explicitly set and not when set to 'auto'.

When set to 'auto' dtype is numpy.float64 array, which is a third behavior. However we are matching what is done by sklearn there.

**Minimal Complete Verifiable Example**:

[see test_basic_array ](https://github.com/dask/dask-ml/blob/7b8c8e0f2d3fb029850e9bf07203b8f7176bdf3e/tests/preprocessing/test_encoders.py#L24)

**Anything else we need to know?**:

This is illustrated in the [failing test](https://github.com/dask/dask-ml/actions/runs/4083895663/jobs/7039952888#step:5:1050)

Sklearn documentation found [here](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.OneHotEncoder.html) shows dtype as object where dask-ml shows dtype as `<U1` found [here](https://github.com/dask/dask-ml/blob/80eb0d81b22df3762706384c3ca1c0be4137e591/dask_ml/preprocessing/_encoders.py#L104)

**Environment**:

- Dask version:  dask-ml-3.8 conda env  dask 2023.1.1
- Python version: 3.8, 3.9, 3.10
- Operating System:  ubuntu-latest
- Install method (conda, pip, source): conda

**Comment by narnia24** [comment]
hey can i work on this?

**Comment by ruthvik0612** [comment]
Hi 👋  
I’m a new contributor and a recent graduate working with ML preprocessing tools.  
I’m interested in working on this issue to understand and align (or document) the behavior of OneHotEncoder with text labels between dask-ml and scikit-learn.  

Before starting, I wanted to ask:
- Is this issue still valid?
- Would you prefer a behavioral fix, additional tests, or documentation clarification?

Thanks!

**Comment by TomAugspurger** [comment]
Thanks @ruthvik0612. I believe this was fixed by https://github.com/dask/dask-ml/pull/965/changes.
